compaction/compaction_manager.cc

```diff
@@ -548,7 +548,9 @@ inline bool compaction_manager::maybe_stop_on_error(future<> f, stop_iteration w
         // case we won't retry.
         retry = e.retry();
         decision_msg = !retry ? stop_msg : decision_msg;
-        cmlog.info("compaction info: {}: {}", e.what(), decision_msg);
+        auto error_msg = sstring(e.what());
+        logging::log_level level = error_msg.find("failed") == sstring::npos ? logging::log_level::info : logging::log_level::error;
+        cmlog.log(level, "compaction info: {}: {}", error_msg, decision_msg);
```

Comment:
Pattern-matching the error message looks pretty fragile to me.
I'd prefer a more explicit API, maybe add a new flag.

Reply:
yeah, it's should easy to complement the `_retry` flag.
Maybe turn it into an enum class.
(Reminds me the good old [Abort, Retry, Fail?](https://en.wikipedia.org/wiki/Abort,_Retry,_Fail%3F) days :-) )

Reply:
If we have an exception, shouldn't we use error unconditionally?

If it's an expected exception, catch it (by type) and swallow it.

Reply:
@avikivity we (ab)use the stop_compaction_exception for several purposes.  We can split it into 3 different exceptions that can be caught and handled by type.

@raphaelsc what do you think?